### DISCUSSION TITLE
I want to modify the dict file in espeak-ng-data

### DISCUSSION THREAD

**Discussion by harmlessman** [espeak-ng/espeak-ng#2108] [Q&A] [answered]
Hello,

I am using phonemizer to convert text in my language into IPA, with espeak-ng as the backend. The issue is that there are cases where the IPA conversion behaves unexpectedly, and I want to correct them.

To do this, I need to modify the dict file in espeak-ng-data. However, I am unsure how to properly read and edit the file. Additionally, I would like to know how I can contribute these modifications back to the project after making them.

If anyone can provide guidance on this matter, I would greatly appreciate your help.

Thank you!

**Comment by fabiolimace** [comment, marked as answer]
The transcription rules are in the `dictsource/XX_rules` file, where XX is the language code. The rules are described using the [Kirshenbaum](https://en.wikipedia.org/wiki/Kirshenbaum) system.

The IPA symbols associated with each phoneme (or Kirshenbaum symbol) is in the `phsource/XX_language` file.

After making the corrections, you have to compile the language again. Run this command inside of `dictsource` directory:

```bash
sudo espeak-ng --path=/usr/share/espeak-ng-data/ --compile=en
```
However, in order to understand the files' structure and syntax, it's recommended studying the documentation.
